engine/new_runtime/src/locked_memory.rs

```diff
@@ -0,0 +1,115 @@
+use crate::crypto_utils::crypto_box::{BoxProvider, Key};
+use crate::memories::buffer::Buffer;
+use crate::types::Bytes;
+use core::fmt::Debug;
+use zeroize::Zeroize;
+
+#[derive(Debug)]
+pub enum MemoryError {
+    EncryptionError,
+    DecryptionError,
+    SizeNeededForAllocation,
+    ConfigurationNotAllowed,
+    FileSystemError,
+}
+
+#[derive(Debug)]
+pub enum ProtectedConfiguration {
+    ZeroedConfig(),
+    BufferConfig(usize),
+}
+
+// Different possible configuration for the memory
+// This is still in development
+// Size is an Option type because size is required for memory allocation but
+// not necessary for unlocking the memory
+#[derive(Debug)]
+pub enum LockedConfiguration<P: BoxProvider> {
```

Comment:
> Yea that was the first idea but I was thinking that if we require other params in the future it would be cleaner to add it in the config rather than putting everything as an option in the function call

Good point. Yes if we add more properties than it might make sense. Maybe we can leave it as it is for now and once we have the full implementation and integration we can revive this discussion.

> `ZeroedConfig` is used when zeroing the memory it's a bit like a default value.

But do we need the config as a property in `EncryptedRam` / `FileMemory` at all? The only useful information it holds is whether the data is encrypted or not, but that could just be set as a bool, no? Because we replace the `Key` itself with a random value anyway, and also the `size` information is not used afterwards anymore. If we remove it, then also the "default" `ZeroedConfig` wouldn't be needed anymore.